### PR TITLE
Add wet snow friction type [com1DFA]

### PR DESCRIPTION
**Pull request by awirb**
Here is the option to add an enthalpy-dependent friction angle (and a diffusion force).

https://hedgedoc.avaframe.org/UrhmNu-4QN6Uh0lPG6gU_g#

### CODE REVIEW

File: avaframe/com1DFA/DFAfunctionsCython.pyx [hunk 1]
```diff
@@ -344,6 +389,8 @@ cpdef (double, double) computeEntMassAndForce(double dt, double entrMassCell,
           # erosion: erode according to shear and erosion energy
           dm = areaPart * tau * uMag * dt / entEroEnergy
           areaEntrPart = areaPart
+          # TODO why not this?
+          #areaEntrPart = math.sqrt(areaPart) * uMag * dt
```
Comment (fso42) [inline avaframe/com1DFA/DFAfunctionsCython.pyx:393]: Should this be in here?

File: avaframe/com1DFA/com1DFACfg.ini [hunk 2]
```diff
@@ -286,6 +286,25 @@ B = 4.13
...
+enthRef = 20000
+# initial temperature in °C
+TIni = -10
+# initial temperature for entrainment in °C
+entTempRef = -10.0
+# J/kgK (ice)
+cpIce = 2050.0
+# ToDo: Not used
+# J/kgK (water)
+cpWtr = 4187.0
```
Comment (fso42) [inline avaframe/com1DFA/com1DFACfg.ini:302]: Let's just remove this here. And keep the comment in com1DFA.py. If that is ever reused, we can add this parameter back in...

File: avaframe/tests/test_com1DFA.py [hunk 3]
```diff
@@ -575,6 +575,9 @@ def test_initializeMassEnt():
     dem['header']['xllcenter'] = 0.0
     dem['header']['yllcenter'] = 0.0
 
+    cfg = configparser.ConfigParser()
+    cfg['GENERAL'] = {'rhoEnt': '200.0', 'entTempRef': '-10.', 'cpIce': '2050.', 'Tini': '-10.'}
```
Comment (fso42) [inline avaframe/tests/test_com1DFA.py:579]:
```suggestion
    cfg['GENERAL'] = {'rhoEnt': '200.0', 'entTempRef': '-10.', 'cpIce': '2050.', 'TIni': '-10.'}
```

File: avaframe/tests/test_com1DFA.py [hunk 4]
```diff
@@ -896,7 +899,8 @@ def test_releaseSecRelArea():
     cfg = configparser.ConfigParser()
     cfg['GENERAL'] = {'rho': '200.', 'gravAcc': '9.81', 'massPerParticleDeterminationMethod': 'MPPDH',
                       'interpOption': '2', 'sphKernelRadius': '1', 'deltaTh': '0.25', 'seed': '12345',
-                      'initPartDistType': 'uniform', 'thresholdPointInPoly': '0.001', 'avalancheDir': 'data/avaTest'}
+                      'initPartDistType': 'uniform', 'thresholdPointInPoly': '0.001', 'avalancheDir': 'data/avaTest',
+                      'entTempRef': '-10.', 'cpIce': '2050.', 'Tini': '-10.'}
```
Comment (fso42) [inline avaframe/tests/test_com1DFA.py:903]:
```suggestion
                      'entTempRef': '-10.', 'cpIce': '2050.', 'TIni': '-10.'}
```

File: avaframe/tests/test_com1DFA.py [hunk 5]
```diff
@@ -1033,7 +1038,8 @@ def test_initializeParticles():
     cfg['GENERAL'] = {'resType': 'ppr|pft|pfv', 'rho': '200.', 'gravAcc': '9.81',
                       'massPerParticleDeterminationMethod': 'MPPDH',
                       'interpOption': '2', 'sphKernelRadius': '1', 'deltaTh': '0.25', 'seed': '12345',
-                      'initPartDistType': 'uniform', 'thresholdPointInPoly': '0.001', 'avalancheDir': 'data/avaTest'}
+                      'initPartDistType': 'uniform', 'thresholdPointInPoly': '0.001', 'avalancheDir': 'data/avaTest',
+                      'entTempRef': '-10.', 'cpIce': '2050.', 'Tini': '-10.'}
```
Comment (fso42) [inline avaframe/tests/test_com1DFA.py:1042]:
```suggestion
                      'entTempRef': '-10.', 'cpIce': '2050.', 'TIni': '-10.'}
```

File: avaframe/tests/test_com1DFA.py [hunk 6]
```diff
@@ -1495,7 +1501,8 @@ def test_initializeSimulation(tmp_path):
                       'rho': '200.', 'gravAcc': '9.81', 'massPerParticleDeterminationMethod': 'MPPDH',
                       'interpOption': '2', 'sphKernelRadius': '1', 'deltaTh': '0.25', 'seed': '12345',
                       'initPartDistType': 'uniform', 'thresholdPointInPoly': '0.001', 'avalancheDir': 'data/avaTest',
-                      'dRes': '0.3', 'cw': '0.5', 'sres': '5', 'initialiseParticlesFromFile': 'False'}
+                      'dRes': '0.3', 'cw': '0.5', 'sres': '5', 'initialiseParticlesFromFile': 'False',
+                      'entTempRef': '-10.', 'cpIce': '2050.', 'Tini': '-10.'}
```
Comment (fso42) [inline avaframe/tests/test_com1DFA.py:1505]:
```suggestion
                      'entTempRef': '-10.', 'cpIce': '2050.', 'TIni': '-10.'}
```

**Review by fso42 — CHANGES_REQUESTED** [review]
@awirb: how do you want to handle the documentation for wet snow? In  a separate PR?

File: avaframe/com1DFA/com1DFACfg.ini (hunk 2 above)
Comment (awirb) [reply avaframe/com1DFA/com1DFACfg.ini:302]: unused parameters removed

File: avaframe/com1DFA/DFAfunctionsCython.pyx (hunk 1 above)
Comment (awirb) [reply avaframe/com1DFA/DFAfunctionsCython.pyx:393]: not solved yet

File: avaframe/com1DFA/com1DFACfg.ini [hunk 7]
```diff
@@ -285,6 +285,21 @@ R = 0.05
 B = 4.13
 #+++++++++++++Voellmy friction model
 xsi = 4000.
+#++++++++++Wet snow friction type
+# if wetSnow is chosen, a Voellmy-type friction model with enthalpy dependent mu is used
+# set frictModel to wetSnow in the friction model setting
+# starting mu0 value
+mu0WetSnow = 0.19981971769923678816664595
```
Comment (fso42) [inline avaframe/com1DFA/com1DFACfg.ini:292]: lets set this to 0.2

**Review by fso42 — CHANGES_REQUESTED** [review]

**Review by fso42 — APPROVED** [review]

**Comment by fso42** [comment]
Standardtests ok

**Pull request merged**